File: 필수문항/148_박터뜨리기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        int K = Integer.parseInt(st.nextToken());
        int minimum = K * (K + 1) / 2;
        if (N < minimum) {
            System.out.println(-1);
        } else {
            N = (N - minimum) % K;
            if (N == 0) {
                System.out.println(K - 1);
            } else {
                System.out.println(K);
            }
        }
    }
}

File: 필수문항/148_박터뜨리기.py
import sys

input = sys.stdin.readline

N, K = map(int, input().split())
minimum = K * (K + 1) / 2
if N < minimum:
    print(-1)
else:
    N = (N - minimum) % K
    if N == 0:
        print(K - 1)
    else:
        print(K)
